### PR TITLE
[Update independent.py]: add explicit string representation

### PR DESCRIPTION
**Pull request by zuoxingdong**

### CODE REVIEW

File: torch/distributions/independent.py [hunk 1]
```diff
@@ -96,3 +96,6 @@ def enumerate_support(self, expand=True):
         if self.reinterpreted_batch_ndims > 0:
             raise NotImplementedError("Enumeration over cartesian product is not implemented")
         return self.base_dist.enumerate_support(expand=expand)
+
+    def __repr__(self):
+        return '{}({})'.format(self.__class__.__name__, self.base_dist)
```
[the remark this answers is not in the document]
Comment (apaszke) [reply torch/distributions/independent.py]: It should be ok.

[the remark this answers is not in the document]
Comment (zuoxingdong) [reply torch/distributions/independent.py]: Done

File: torch/distributions/independent.py [hunk 2]
```diff
@@ -96,3 +96,7 @@ def enumerate_support(self, expand=True):
         if self.reinterpreted_batch_ndims > 0:
             raise NotImplementedError("Enumeration over cartesian product is not implemented")
         return self.base_dist.enumerate_support(expand=expand)
+
+    def __repr__(self):
+        string = '{}({}, {})'
+        return string.format(self.__class__.__name__, self.reinterpreted_batch_dims, self.base_dist)
```
Comment (apaszke) [inline torch/distributions/independent.py]: You put the arguments in an incorrect order.

Comment (ezyang) [inline torch/distributions/independent.py]:
```suggestion
        return string.format(self.__class__.__name__, self.base_dist, self.reinterpreted_batch_dims)
```

**Review by zuoxingdong — COMMENTED** [review]
@ezyang @apaszke Now the ordering has been fixed.

**Comment by zuoxingdong** [comment]
@ezyang @apaszke Not sure why some tests are failed, but doesn't seem like caused by the change in `__repr__`

**Pull request closed without merging**